app/services/checkout/stripe_redirect.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Provides the redirect path if a redirect to the payment gateway is needed
+module Checkout
+  class StripeRedirect
+    def initialize(params, order)
+      @params = params
+      @order = order
+    end
+
+    # Returns the path to the authentication form if a redirect is needed
+    def path
+      return unless stripe_payment_method?
+
+      payment = @order.pending_payments.last
+      return unless payment&.checkout?
+
+      payment.authorize!
+      raise unless payment.pending?
```

Comment:
ah, interesting, I see what you mean. We would have an authentication, then authorization and then confirmation.
But I wanted to keep it as close as possible to the current Spree model, we would need to add authentication to Spree::Payment::Processing. I just went for authrozation.
Not only this, the API reaction to this is pretty good! This authorization request does not fail if the card needs authentication, the API response is a nice "status = needs_action" with a redirect link where I can send the user to for "authentication". And after that I can simply confirm the payment intent.

So, I think it would be more correct to make an "authentication" call (would that be a different API endpoint?) and then an authorization one but I think we are ok as is :+1:

Do you see a scenario where this would not work?

Reply:
thanks a lot for you review btw!

Reply:
> This authorization request does not fail if the card needs authentication, the API response is a nice "status = needs_action" with a redirect link where I can send the user to for "authentication". And after that I can simply confirm the payment intent (this part is done on the normal order payment workflow).

Oh, this is great. :slightly_smiling_face: Nice, @luisramos0.

> Do you see a scenario where this would not work?

No, I'm now happy with this PR. :slightly_smiling_face:

Reply:
thanks!!!